src/modding/ModLoader.cs

```diff
@@ -0,0 +1,278 @@
+            LoadMod(load);
+            loadedMods.Add(load);
+        }
+
+        GD.Print("Mod loading finished");
+    }
+
+    private void LoadMod(string name)
+    {
+        var info = LoadModInfo(name);
+
+        if (info == null)
+        {
+            GD.PrintErr("Can't load mod due to failed info reading: ", name);
+            return;
+        }
+
+        bool loadedSomething = false;
+
+        if (!string.IsNullOrEmpty(info.Info.PckToLoad))
+        {
+            LoadPckFile(Path.Combine(info.Folder, info.Info.PckToLoad));
+            loadedSomething = true;
+        }
+
+        if (!string.IsNullOrEmpty(info.Info.ModAssembly))
+        {
+            Assembly modAssembly;
+            try
+            {
+                modAssembly = LoadCodeAssembly(Path.Combine(info.Folder, info.Info.ModAssembly));
+            }
+            catch (Exception e)
+            {
+                GD.PrintErr("Could not load mod assembly due to exception: ", e);
+                return;
+            }
+
+            if (!CreateModInstance(name, info, modAssembly))
+                return;
+
+            loadedSomething = true;
+        }
+
+        if (!loadedSomething)
+        {
+            GD.Print("A mod contained no loadable resources");
+        }
+    }
+
+    private void UnLoadMod(string name)
+    {
+        var info = LoadModInfo(name);
+
+        if (info == null)
+        {
+            GD.PrintErr("Can't load mod due to failed info reading: ", name);
+            return;
+        }
+
+        if (loadedModAssemblies.TryGetValue(name, out var mod))
+        {
+            GD.Print("Unloaded mod contained an assembly, sending it the unload method call");
+
+            try
+            {
+                if (!mod.Unload())
+                {
+                    GD.PrintErr("Mod's (", name, ") assembly unload method call failed");
+                }
+            }
+            catch (Exception e)
+            {
+                GD.PrintErr("Mod's (", name, ") assembly unload method call failed with an exception: ", e);
+            }
+
+            loadedModAssemblies.Remove(name);
+        }
+    }
+
+    private void RunCodeModFirstRunCallbacks(IMod mod)
+    {
+        var scene = modInterface.CurrentScene;
+
+        mod.CanAttachNodes(scene);
+    }
+
+    private bool CreateModInstance(string name, FullModDetails info, Assembly assembly)
+    {
+        var className = info.Info.AssemblyModClass;
+
+        var type = assembly.GetTypes().FirstOrDefault(t => t.Name == className);
+
+        if (type == null)
+        {
+            GD.Print("No class with name \"", className, "\" found, can't finish loading mod assembly");
+            return false;
+        }
+
+        try
+        {
+            var mod = (IMod)Activator.CreateInstance(type);
+
+            if (!mod.Initialize(modInterface, info.Info))
+            {
+                GD.PrintErr("Mod's (", name, ") initialize method call failed");
```

Comment:
shouldn't this `return false` here?

Reply:
I didn't see a point in that, the ultimate goal is to just collect these errors and show them in a popup to the player, so at this point it's just fine to keep running even with the failure.